src/audio-worklet/migration/worklet-recorder/app.js

```diff
@@ -11,13 +11,25 @@ const context = new AudioContext();
 // Make the visulization more clear to the users
 const WAVEFROM_SCALE_FACTOR = 5
 let isRecording = false;
-let visualizationEnabled = true;
+let initCount = 0;
```

Comment:
Ditto.

Reply:
The reason why I use count instead of boolean is that, we need to control the first time init is sending the "message: 'UPDATE_RECORDING_STATE',  to processor. Since the workflow is: "Click start recording" -> isInitied = true -> Then we don't have another control variable or using other control variable to control  the message posting to audioworklet processor. Therefore, I use initCount as a number to control these two steps. Sorry for the confusion. Maybe I should use a better naming?

Reply:
Don't be sorry! You're doing great here.

If we need to handle more than a binary state in the code, it's better to use the enumeration to make it more explicit. How many states do we need in the SPN example?

Reply:
Thanks for the advices! I split SPN recording state into three:
const recordingStates = {
  notInitialized: 'notInitialized',
  startIniAndRecording: 'startIniAndRecording',
  finishRecording: 'finishRecording'
};

And Worklet recording state into four since Worklet has one more state of sending message to recording-processor:
const recordingStates = {
  notInitialized: 'notInitialized',
  startInitialized: 'startInitialized',
  startRecording: 'startRecording',
  finishRecording: 'finishRecording'
};